### PR TITLE
[type: refactor] refactor abstractshenyuplugin#execute

### PR DESCRIPTION
**Pull request by 847850277**
Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/AbstractShenyuPlugin.java [hunk 1]
```diff
@@ -79,76 +79,72 @@ public abstract class AbstractShenyuPlugin implements ShenyuPlugin {
...
+                if (matchCacheConfig.getSelectorEnabled() && matchSelectorData.getLeft()) {
+                    selectorData = new SelectorData();
+                    selectorData.setPluginName(pluginName);
+                    cacheSelectorData(path, selectorData);
                 }
+                return handleSelectorIfNull(pluginName, exchange, chain);
             } else {
-                if (StringUtils.isBlank(selectorData.getId())) {
-                    return handleSelectorIfNull(pluginName, exchange, chain);
+                if (matchCacheConfig.getSelectorEnabled() && matchSelectorData.getLeft()) {
```
Comment (yu199195) [inline shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/AbstractShenyuPlugin.java:109]: other alreay cache，  if  StringUtils.isBlank(selectorData.getId()， mybe return~

Comment (847850277) [reply shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/AbstractShenyuPlugin.java:109]: Preprocessing then returns.

image

File: shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/AbstractShenyuPlugin.java [hunk 2]
```diff
@@ -79,76 +79,72 @@ public abstract class AbstractShenyuPlugin implements ShenyuPlugin {
...
+        final String pluginName = named();
+        final String path = exchange.getRequest().getURI().getPath();
         PluginData pluginData = BaseDataCache.getInstance().obtainPluginData(pluginName);
-        if (Objects.nonNull(pluginData) && pluginData.getEnabled()) {
-            final String path = exchange.getRequest().getURI().getPath();
-            SelectorData selectorData = obtainSelectorDataCacheIfEnabled(exchange);
+        List<SelectorData> selectors = BaseDataCache.getInstance().obtainSelectorData(pluginName);
+        SelectorData selectorData = obtainSelectorDataCacheIfEnabled(path);
+        // early exit
+        if (Objects.isNull(pluginData) || !pluginData.getEnabled()) {
```
[the remark this answers is not in the document]
Comment (847850277) [reply shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/AbstractShenyuPlugin.java:85]: fixed it

File: shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/AbstractShenyuPlugin.java [hunk 3]
```diff
@@ -79,76 +79,72 @@ public abstract class AbstractShenyuPlugin implements ShenyuPlugin {
...
+                if (matchCacheConfig.getSelectorEnabled() && matchSelectorData.getLeft()) {
+                    selectorData = new SelectorData();
+                    selectorData.setPluginName(pluginName);
+                    cacheSelectorData(path, selectorData);
                 }
+                return handleSelectorIfNull(pluginName, exchange, chain);
             } else {
-                if (StringUtils.isBlank(selectorData.getId())) {
-                    return handleSelectorIfNull(pluginName, exchange, chain);
+                if (matchCacheConfig.getSelectorEnabled() && matchSelectorData.getLeft()) {
```
Comment (yu199195) [inline shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/AbstractShenyuPlugin.java:109]: alreay cache，not again cache

Comment (847850277) [reply shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/AbstractShenyuPlugin.java:109]: this is from From the old logic, seem No duplicate cache appears

image

image

**Review by yu199195 — APPROVED** [review]

**Pull request merged**